kanidmd/src/lib/be/dbvalue.rs

```diff
@@ -70,41 +87,64 @@ pub struct DbCredV1 {
 
 #[derive(Serialize, Deserialize, Debug)]
 pub struct DbValueCredV1 {
-    pub t: String,
-    pub d: DbCredV1,
+    #[serde(rename = "t")]
+    pub tag: String,
+    #[serde(rename = "d")]
+    pub data: DbCredV1,
 }
 
 #[derive(Serialize, Deserialize, Debug)]
 pub struct DbValueTaggedStringV1 {
-    pub t: String,
-    pub d: String,
+    #[serde(rename = "t")]
+    pub tag: String,
+    #[serde(rename = "d")]
+    pub data: String,
 }
 
 #[derive(Serialize, Deserialize, Debug)]
 pub struct DbValueEmailAddressV1 {
-    pub d: String,
+    #[serde(rename = "d")]
+    pub email_addr: String,
```

Comment:
d for data here I think, email address is already clear from the struct name.

Reply:
Why not use a tuple struct then? (I'm not sure how this would effect serialization though.)